kms/asymmetric.go

```diff
@@ -0,0 +1,260 @@
+// Copyright 2018 Google Inc. All rights reserved.
+// Use of this source code is governed by the Apache 2.0
+// license that can be found in the LICENSE file.
+
+// Samples for asymmetric keys feature of Cloud Key Management Service: https://cloud.google.com/kms/
+package kms
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/asn1"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"log"
+	"math/big"
+
+	cloudkms "cloud.google.com/go/kms/apiv1"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+// [START kms_create_asymmetric_key]
+
+// createAsymmetricKey creates a new RSA encrypt/decrypt key pair on KMS.
+// example keyRingName: "projects/PROJECT_ID/locations/global/keyRings/RING_ID"
+func createAsymmetricKey(keyRingName, keyId string) error {
+	ctx := context.Background()
+	client, err := cloudkms.NewKeyManagementClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Build the request.
+	req := &kmspb.CreateCryptoKeyRequest{
+		Parent:      keyRingName,
+		CryptoKeyId: keyId,
+		CryptoKey: &kmspb.CryptoKey{
+			Purpose: kmspb.CryptoKey_ASYMMETRIC_DECRYPT,
+			VersionTemplate: &kmspb.CryptoKeyVersionTemplate{
+				Algorithm: kmspb.CryptoKeyVersion_RSA_DECRYPT_OAEP_2048_SHA256,
+			},
+		},
+	}
+	// Call the API.
+	result, err := client.CreateCryptoKey(ctx, req)
+	if err != nil {
+		return err
+	}
+	log.Printf("Created crypto key. %s", result)
+	return nil
+}
+
+// [END kms_create_asymmetric_key]
+
+// [START kms_get_asymmetric_public]
+
+// getAsymmetricPublicKey retrieves the public key from a saved asymmetric key pair on KMS.
+// example keyName: "projects/PROJECT_ID/locations/global/keyRings/RING_ID/cryptoKeys/KEY_ID/cryptoKeyVersions/1""
+func getAsymmetricPublicKey(keyName string) (interface{}, error) {
```

Comment:
The `get` prefix is usually discouraged in Go. But, I think this is OK if you think it's the best name. https://golang.org/doc/effective_go.html#Getters

Reply:
I think it would be best to keep get. We want the name to be consistent with other languages, and this isn't exactly a getter anyway